app/models/adapters/payment_history_adapter.rb

```diff
@@ -13,7 +13,7 @@ def initialize(input)
       else
         @input_payments = [@input_payment].flatten
         payments, return_payments = @input_payments.partition do |payment|
-          payment.dig(:return_payment, :check_trace_number).blank?
+          payment&.dig(:return_payment, :check_trace_number).blank?
```

Comment:
The errors say it's happening on line 8. I wonder why... 😕 
Error:
```
adapters/payment_history_adapter.rb:8:in `initialize': undefined method `dig' for nil (NoMethodError)
```

Line 8:
```ruby
@input_payment = input&.dig(:payments, :payment)
```

Reply:
Line 8 is calling these methods in the file.

Reply:
Payment is null

Reply:
ok. Confusing error message to me, but it makes sense to add `&` where you did.